sentry-ruby/lib/sentry/transport.rb

```diff
@@ -7,20 +7,36 @@ class Transport
     USER_AGENT = "sentry-ruby/#{Sentry::VERSION}"
 
     attr_accessor :configuration
+    attr_reader :rate_limits
 
     def initialize(configuration)
       @configuration = configuration
       @transport_configuration = configuration.transport
       @dsn = configuration.dsn
+      @rate_limits = {}
     end
 
     def send_data(data, options = {})
       raise NotImplementedError
     end
 
     def send_event(event)
+      event_hash = event.to_hash
+      event_type = event_hash[:type] || event_hash['type']
```

Comment:
Is this the `type` coming from the event payload? I'm asking because `event` is not a valid event type.

For reference, the valid types are: `error`, `csp`, `hpkp`, `expectct`, `expectstaple`, `transaction`. I would propose that you check for `transaction`, and in all other cases fall back to `error`. 

Please ensure that `event.type: event` is not sent to the server.

Reply:
wow this is the first time I learn this. where is this documented? the closest thing I find right now is in the [schema repo](https://github.com/getsentry/sentry-data-schemas/blob/main/relay/event.schema.json#L346-L347) (through [this document](https://develop.sentry.dev/sdk/event-payloads/types/)). but that doesn't exactly match watch you mentioned here.

Reply:
https://github.com/getsentry/relay/pull/954

Reply:
so to summarize:
- all transactions should have a field `event_type: "transaction"`
- events should not have `event_type` field
- when rate limiting, events should be mapped with `error` category regardless if the event actually contains an  error

Reply:
The field should be `type`, otherwise that's correct 👍 

> when rate limiting, events should be mapped with error category regardless if the event actually contains an error

Yes. This is a simplification of the actual system, but it is correct enough and we'll also document it soon this way. Just ensure that `type: "transaction"` events are mapped with their own category.

Reply:
thanks for the confirmation, I've implemented the type changes in a separate PR: https://github.com/getsentry/sentry-ruby/pull/1354